geomesa-web/geomesa-web-security/src/main/scala/org/locationtech/geomesa/web/security/GMSecuredDataFactory.scala

```diff
@@ -0,0 +1,91 @@
+package org.locationtech.geomesa.web.security
+
+import com.google.common.collect.Maps
+import org.apache.accumulo.core.security.{Authorizations, ColumnVisibility, VisibilityEvaluator}
+import org.geoserver.security.WrapperPolicy
+import org.geoserver.security.decorators.{DecoratingFeatureSource, DefaultSecureDataFactory}
+import org.geotools.data._
+import org.geotools.data.simple.{SimpleFeatureCollection, SimpleFeatureIterator}
+import org.geotools.feature.collection.{FilteringSimpleFeatureCollection, FilteringSimpleFeatureIterator}
+import org.geotools.feature.{DefaultFeatureCollection, FeatureCollection}
+import org.locationtech.geomesa.core.security.SecurityUtils
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+import org.opengis.filter.{Filter, FilterVisitor}
+import org.springframework.security.core.context.SecurityContextHolder
+
+class GMSecuredDataFactory extends DefaultSecureDataFactory {
+
+  override def secure(o: scala.Any, policy: WrapperPolicy): AnyRef =
+    super.secure(o, policy) match {
+      case ds: DataStore =>
+        new GMSecureDataStore(ds)
+
+      case fs: FeatureSource[SimpleFeatureType, SimpleFeature] =>
+        new GMSecureFeatureSource(fs)
+
+      case fc: SimpleFeatureCollection =>
+        new GMSecureFeatureCollection(fc)
+
+      case so =>
+        so
+    }
+
+
+  override def getPriority: Int = 1
+}
+
+object GMSecuredDataFactory {
+  import scala.collection.JavaConversions._
+
+  def getAuthorizations: Authorizations = {
+    val auths = SecurityContextHolder.getContext.getAuthentication.getAuthorities.map(_.getAuthority).toList
```

Comment:
I'm not sure it does - at least for the short term.  Basically, we're only using this in GeoServer so we can assume that auths are set via the Spring Security thread local.  It's already pluggable there, this code just needs to extract it out of the thread local and use it.